src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +284,165 @@ def get_info(
+        self._validate_run_task_request(request)
+
+        with self._tasks_lock:
+            if request.task_id in self._running_tasks:
+                context.abort(
+                    grpc.StatusCode.ALREADY_EXISTS,
+                    f"Task {request.task_id} already exists",
+                )
+
+            task_execution = _TaskExecution(request)
+            self._running_tasks[request.task_id] = task_execution
+
+            task_execution.thread = threading.Thread(
+                target=self._execute_task_in_thread,
+                args=(request, task_execution),
+                daemon=True,
+            )
+            task_execution.thread.start()
+
+        return StartTaskResponse(task_id=request.task_id)
+
+    def await_task(
+        self, request: AwaitTaskRequest, context: grpc.ServicerContext
+    ) -> Generator[TaskStreamResponse, None, None]:
+        """Wait for task completion and stream progress updates."""
+        task_id = request.task_id
+
+        with self._tasks_lock:
+            task_execution = self._running_tasks.get(task_id)
+
+        if task_execution is None:
+            context.abort(
+                grpc.StatusCode.NOT_FOUND,
+                f"Task {task_id} not found",
+            )
+
+        # Stream progress updates until task completes
+        response = None
+        while True:
+            with task_execution.updated:
+                if (
+                    task_execution.response.outcome_code
+                    != TaskOutcomeCode.TASK_OUTCOME_CODE_UNKNOWN
+                ):
+                    response = task_execution.response
+                    break
+
+                task_execution.updated.wait()
+                progress = task_execution.progress
+
+            if progress:
+                yield TaskStreamResponse(progress=progress)
+
+        yield TaskStreamResponse(task_result=response)
+
+    def delete_task(
+        self, request: DeleteTaskRequest, context: grpc.ServicerContext
+    ) -> Empty:
+        """Delete a task and clean up resources."""
+        task_id = request.task_id
+
+        with self._tasks_lock:
+            task_execution = self._running_tasks.get(task_id)
+
+            if task_execution is None:
+                context.abort(
+                    grpc.StatusCode.NOT_FOUND,
+                    f"Task {task_id} not found",
+                )
+
+            with task_execution.updated:
+                if (
+                    task_execution.response.outcome_code
+                    == TaskOutcomeCode.TASK_OUTCOME_CODE_UNKNOWN
+                ):
+                    task_execution.response.outcome_code = (
+                        TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE
+                    )
+                    # TODO: Consider adding a Cancelled reason here
+                    task_execution.response.failure_reason = (
+                        TaskFailureReason.TASK_FAILURE_REASON_FUNCTION_ERROR
+                    )
```

Comment:
If we delete unfinished task we should just return error. We can't stop a running task in FE cause it might be running customer code

Reply:
The task appears to go away from Executor PoV after delete_task RPC finished successfully. If delete_task blocks forever or Executor client deadline exceeded then it didn't delete successfully.

Reply:
Done